src/components/TextInput/TextInput.tsx

```diff
@@ -205,6 +207,8 @@ const TextInput = ({
     debounceInterval,
   ]);
 
+  const internalInputRef = useRef(inputRef?.current);
```

Comment:
Not sure this is the right approach or that my understanding of refs is 100%. I deleted the test checking for inputRef.current because this change causes that to fail.

Reply:
I think you can just add a new ref and merge the two refs (there's a utility function called mergeRefs for this, check out how it's used elsewhere)